diag_manager/fms_diag_object.F90

```diff
@@ -396,17 +422,323 @@ function diag_obj_is_static (obj) result (rslt)
     rslt = obj%static
 end function diag_obj_is_static
 
-function diag_obj_get_vartype (obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    integer :: rslt
-    rslt = obj%vartype
-end function diag_obj_get_vartype
-
-function diag_obj_get_varname(obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    character(len=len(obj%varname)) :: rslt
-    rslt = obj%varname
-end function diag_obj_get_varname
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! Get functions
+
+!> @brief Gets metedata
+!! @return metadata string array, or a single space if metadata is not allocated
```

Comment:
As a group, we decided to return a copy, and all of the comments of the `get_` functions have been updated to reflect that we are returning a copy.  

We can't return an unallocated.  Returning an unallocated will at best cause a crash and at worst not crash and do some sort of compiler undefined behavior.

We can't initialize the strings because we would have to limit their size if we did.  These have to be allocatable.

The `has_` functions are coming in a future separate PR to not make this PR too large.